src/ui/pages/work/activities/dialogs/AddActivityDialog.js

```diff
@@ -149,7 +166,7 @@ function AddActivityDialog(props) {
                                 value={formValues['taskCode']}
                                 valueDesc={formValues['taskDesc']}
                                 descKey="taskDesc"
-                                updateProperty={updateFormValues}/>
+                                updateProperty={updateFormValuesFromTaskPlan}/>
```

Comment:
I think `updateFormValues` is more representative of what should be done when this input field is changed. Perhaps `updateFormValuesFromTaskPlan` could be renamed to `onTaskCodeChanged`. This function would then only be responsible for performing the webservice call, changing the form values again if necessary. `updateFormValues` would thus be responsible for calling the `onTaskCodeChanged` when the `taskCode` is changed.